packages/cells/src/widget.ts

```diff
@@ -1040,8 +1051,23 @@ export class CodeCell extends Cell<ICodeCellModel> {
       this.addClass(DIRTY_CLASS);
     }
 
-    this._outputPlaceholder = new OutputPlaceholder(() => {
-      this.outputHidden = !this.outputHidden;
+    this._outputPlaceholder = new OutputPlaceholder({
+      callback: () => {
+        this.outputHidden = !this.outputHidden;
+      },
+      text: (
+        ((this.model.toJSON().outputs as nbformat.IOutput[])[0]
+          ?.text as string) ?? ''
+      ).split('\n')[0]
+    });
+
+    this.model.contentChanged.connect((sender, args) => {
```

Comment:
This is wrong, you should listen for `this.model.outputs.changed`.

Could you also implement the same optimization as the one suggested for the input placeholder (aka only updating the placeholder if it is visible).